app.js

```diff
@@ -1,138 +1,188 @@
+  else {
     recognizing = false;
-    submit_button.style.display = 'none';
-    start_button.style.display = 'inline-block';
+    submit_button.style.display = "none";
+    start_button.style.display = "inline-block";
   }
   const req = new XMLHttpRequest();
   const baseUrl = BASE_URL_VS + "/text/1";
-  const subject_name = document.getElementById('subject_name').value;
-  const lecture_name = document.getElementById('lecture_name').value;
-  const num_lines = document.getElementById('num_lines').value;
-  if(num_lines == null)num_lines = -1;
+  const subject_name = document.getElementById("subject_name").value;
+  const lecture_name = document.getElementById("lecture_name").value;
+  const num_lines = document.getElementById("num_lines").value;
+  if (num_lines == null) num_lines = -1;
   // const transcript = document.getElementById('transcript').value;
   // const transcript_data = localStorage.getItem('current_session');
-  if(is_google_meet){
+  if (is_google_meet) {
     catch_meet = false;
     transcript_data = google_meet_transcript_data;
   }
   const urlParams = {
-    "subject_name":subject_name,
-    "lecture_name":lecture_name,
-    "transcript":transcript_data,
-    "email":(JSON.parse(localStorage.getItem('user')) && JSON.parse(localStorage.getItem('user')).email ) || TEMP_EMAIL, 
-    "images":JSON.parse(localStorage.getItem('images')),
-    "is_gmeet":is_google_meet,
-    "num_lines":num_lines
+    subject_name: subject_name,
+    lecture_name: lecture_name,
+    transcript: localStorage.getItem("meet-data"),
+    email:
+      (JSON.parse(localStorage.getItem("user")) &&
+        JSON.parse(localStorage.getItem("user")).email) ||
+      TEMP_EMAIL,
+    images: JSON.parse(localStorage.getItem("images")),
+    is_gmeet: is_google_meet,
+    num_lines: num_lines,
   };
   console.log(urlParams);
   req.open("POST", baseUrl);
   req.setRequestHeader("Content-Type", "application/json");
   req.send(JSON.stringify(urlParams));
 
-  req.onreadystatechange = function() { // Call a function when the state changes.
-      if (this.readyState === XMLHttpRequest.DONE && this.status === 200) {
-          console.log("Got response 200!");
-          transcript_data = [];
-          localStorage.setItem('images',null);
-      }
-  }
+  req.onreadystatechange = function () {
+    // Call a function when the state changes.
+    if (this.readyState === XMLHttpRequest.DONE && this.status === 200) {
+      console.log("Got response 200!");
+      transcript_data = [];
+      localStorage.setItem("images", null);
+    }
+  };
 
   is_google_meet = false;
 });
 
+stop_button.addEventListener("click", function (e) {
+  e.preventDefault();
+  if (!is_google_meet) toggleSpeechRecognition();
+  else {
+    recognizing = false;
+    stop_button.style.display = "none";
+    resume_button.style.display = "inline-block";
+  }
+
+  if (is_google_meet) {
+    catch_meet = false;
+    transcript_data = google_meet_transcript_data;
+  }
+});
```

Comment:
there is no need to assign the google_meet_transcript_data as it is already stored in the localstorage